pkg/tecdsa/dkg/dkg.go

```diff
@@ -64,7 +64,7 @@ func (e *Executor) Execute(
 
 	registerUnmarshallers(channel)
 
-	preParams, err := e.tssPreParamsPool.Get()
+	preParams, err := e.tssPreParamsPool.GetNow()
 	if err != nil {
 		return nil, 0, fmt.Errorf("failed fetching pre-params: [%v]", err)
```

Comment:
We could also change it to `%w` so that we can unwrap `ErrPoolEmpty` and differentiate if it's an empty pool or e.g. persistence problem if needed.

Reply:
https://github.com/keep-network/keep-core/pull/3207/commits/28eb1c7814703a137bed34e428f752b188dd2c39